pagopa/io-app
# chore: [IOPID-359] First onboarding refactor

## Short description
Include a summary of the changes.

## List of changes proposed in this pull request
- ts/sagas/profile.ts: last_app_version and preferred_languages are always set if not found at every profile upsert. Upsert and checks on the profile are prevented until acceptance of the tos which becomes the first upsert of the app (during this first upsert it also sets the app version and language). This way there are also no more upsert and duplicate checks when creating a new profile.
- ts/store/reducers/profile.ts: Having profile version 0 is not enough to tell if you are an onboarding. The check is now done on service_preferences_settings which is set to legacy until the last onboarding screen.
- *.test.ts: updates

## How to test
Run dev server with first onboarding set to true and make sure that all the described behaviors are respected.
